### PR TITLE
Add `animate` event for some dynamic  filters and other features

### PR DESCRIPTION
**Pull request by finscn**
Some features in this PR are not for common users , just for developers.

* `animate` event
* `toggle` event
* Pause animate
* custom index.css for improving ui
* `opened` option

### CODE REVIEW

**Comment by finscn** [comment]
Some filters needs to reset some properties when it from disable to enable.
e.g. the `time` of shockwave

File: tools/demo/index.html [hunk 1]
```diff
@@ -6,9 +6,9 @@
         <title>PixiJS Filters Demo</title>
         <meta name='viewport' content='width=device-width, initial-scale=1.0'>
         <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/font-awesome/4.7.0/css/font-awesome.min.css" />
-        <link rel="stylesheet" href="index.css" />
         <script src="https://cdnjs.cloudflare.com/ajax/libs/dat-gui/0.6.5/dat.gui.min.js"></script>
         <script src="https://pixijs.download/dev/pixi.min.js"></script>
+        <link rel="stylesheet" href="index.css" />
```
Comment (bigtimebuddy) [inline tools/demo/index.html]: Why this change?

File: tools/demo/index.css [hunk 2]
```diff
@@ -47,4 +47,10 @@ body {
...
     width: 100%;
-}
\ No newline at end of file
+}
+.dg .property-name {
+    width: 46%;
+}
+.dg .c {
+    width: 54%;
+}
```
Comment (bigtimebuddy) [inline tools/demo/index.css]: What is the purpose of these styles?

Comment (finscn) [reply tools/demo/index.css]: If we don't make the first column become bigger , we can't see the full content of some properties's name

File: tools/demo/index.html (hunk 1 above)
Comment (finscn) [reply tools/demo/index.html]: if `index.css` before `dat.gui.min.js` , some css in `index.css` will be ignored.  ( in fast network or use local dat.gui.min.js).
